### ISSUE TITLE
Make client ARIA-compatible

### ISSUE THREAD

**Issue by klemay** [hypothesis/product-backlog#679]
- [ ] The `search` form should [follow these design patterns](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/search.html)
- [ ] The `sort` menu should [follow these design patterns](https://www.w3.org/TR/wai-aria-practices/#radiobutton) 
- [ ]  The `share` button needs `aria-label` [as explained here](https://dev.opera.com/articles/ux-accessibility-aria-label/#accessible-name-calculation)
- [ ] The `user` icon should have a `button` label and the drop-down menu should follow practices for `listbox-collapsible` [https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html)
- [ ] The `toggle` and `Hide Highlights` buttons should use [expanded]

(https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element)
![aria-client-needed](https://user-images.githubusercontent.com/12087705/41170788-bccca588-6b02-11e8-98ad-495fd733f8d2.png)

**Comment by klemay** [comment]
## Making `search` in the client sidebar accessible

- The `form` element needs `role="search"`
- The text `input` should have `type="search"` instead of `type="text"`
- The search `button` should have an `aria-label="Search"`

See: https://github.com/hypothesis/client/pull/741/commits/1df1978497cd398c9dee09c5d09d01c6ac169441

**Comment by klemay** [comment]
## Making `sort` in the client sidebar accessible
- Add `aria-label="Sort annotations"` to `button` element
- Add `aria-haspopup="true"` to `button` element
- Add `role="radiogroup"` to `ul`
- Add `role="radio"` to each `span` within the `li`
- Add `aria-checked` state to each `radio` element (Location is the default option, so that will be set to `true` while the others will be `false`
- Add `aria-labelledby` attribute to associate the `radio` elements with their labels.

**Comment by klemay** [comment]
## Making `share` accessible
- The `a` element needs `aria-label="Share this page"`

(NB: ideally this would be a `button` rather than `a` because displaying the Bouncer link is not a linking action, but adding this `aria-label` will at least allow the screenreader to announce what the `share` icon does.

See: https://github.com/hypothesis/client/pull/741/commits/1fdc7453d5a475e24a3a5ee18fda2a1b05115532

**Comment by klemay** [comment]
This was started on a hack day and never got finished - going to close this and use some of the research here as part of current accessibility initiative, as we will likely break down/prioritize these tasks as individual issues. New epic lives here: https://github.com/hypothesis/product-backlog/issues/753
